errors/errors_task.go

```diff
@@ -93,23 +93,30 @@ func (err *TaskNameFlattenConflictError) Code() int {
 	return CodeTaskNameConflict
 }
 
-// TaskCalledTooManyTimesError is returned when the maximum task call limit is
-// exceeded. This is to prevent infinite loops and cyclic dependencies.
-type TaskCalledTooManyTimesError struct {
+// TaskCyclicExecutionDetectedError is returned when the Execution Graph detects
+// a cyclic execution condition.
+type TaskCyclicExecutionDetectedError struct {
```

Comment:
Suggestion: I'd use `CallCycle` instead of `CyclicExecution` everywhere:

```suggestion
// TaskCallCycleDetectedError is returned when the Execution Graph detects
// a call cycle.
type TaskCallCycleDetectedError struct {
```